### DISCUSSION TITLE
What is the type of the result of `findMany`

### DISCUSSION THREAD

**Discussion by alan345** [prisma/prisma#3355] [Q&A] [answered]
With typescript, I want to use the type given my the query `findMany`. I need to use this type inside the `map()` function

for example, how can I get the type `WHAT_IS_THE_TYPE_FOR_Source` generated by prisma client:
```ts
    const sources: WHAT_IS_THE_TYPE_FOR_Source[] = await ctx.prisma.source.findMany({
      where: args.where,
      skip: args.skip,
      take: args.first,
    })
    return {
      edges: sources.map((singleData: WHAT_IS_THE_TYPE_FOR_Source) => { return { node: singleData } }),
      aggregate: {
        count: await ctx.prisma.source.count({ where: args.where })
      }
    }
```

Thanks!

**Comment by ryands17** [comment, marked as answer]
Hey @alan345 👋 
In what case would you require to explicitly annotate the type here? TypeScript automatically infers the types from what Prisma has generated and you can easily get intellisense and code-completion due to what Prisma generates.

Also, if you want to see the type in VSCode, you can hover or `ctrl+click` the method to view the return type.

**Reply by deanbiltup** [reply]
in the case I am trying to pass the result to a method and the paramater of the method needs to declare the type but when my intellisense shows the type, I am left scratching my head looking at an extremely complicated type when I was hoping for just const result: List<User> or something like that.   how to coerce into a reasonable non-complex type so I can pass it to a method with signature List<User> or User[] or something?

**Comment by alan345** [comment]
hey @ryands17 
thanks for the tip.
Another example when I need the type of a return function:

```
me: async (parent: any, args: any, ctx: Ctx, info: any): Promise<User> => {
    const id = await utils.getUserId(ctx)
    return ctx.prisma.user.findOne({ where: { id } })
}
```

**Reply by libraryunbound** [reply]
And how to handle functional differences with the result.  Am I working with an array like [ ] or an Array<T>?

**Reply by twxia** [reply]
Please check https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types related articles

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
